core/network/ipaddress.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network
+
+import (
+	"encoding/binary"
+	"net/netip"
+)
+
+// IPAddress wraps a [netip.Addr] instance and provides a
+// method to decompose the address into msb, lsb uint64 values.
+type IPAddress struct {
+	netip.Addr
+}
+
+// AsInts returns the MSB and LSB uint64 values for the specified address.
+func (a IPAddress) AsInts() (msb uint64, lsb uint64) {
+	addrB := a.AsSlice()
+	if a.Is4() {
```

Comment:
Small nit. This is a slightly contrived so you decided if you want to do this. I think we should tie the branching logic to the size of the slice. It helps to avoid a potential panic and also makes this more thread safe.